tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/repopulate_person_post_deid_test.py

```diff
@@ -0,0 +1,122 @@
+  ({{sex_at_birth_concept_id}}, "sex"),
+  ({{sex_female_concept_id}}, "female"),
+  ({{sex_female_source_concept_id}}, "female_src")
+"""),
+    Template("""
+INSERT INTO `{{project_id}}.{{dataset_id}}.person` (person_id)
+VALUES (1), (2)
+"""),
+    Template("""
+INSERT INTO `{{project_id}}.{{dataset_id}}.observation` (person_id, observation_id, observation_source_concept_id, value_as_concept_id, value_source_concept_id)
+VALUES
+  (1, 100, {{gender_concept_id}}, {{gender_nonbinary_concept_id}}, {{gender_nonbinary_source_concept_id}}),
+  (1, 101, {{sex_at_birth_concept_id}}, {{sex_female_concept_id}}, {{sex_female_source_concept_id}})
+""")
+]
+
+
+class RepopulatePersonPostDeidTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def assertPersonFields(self, person, want):
+        for (k, v) in want.items():
+            self.assertIn(k, person)
+            self.assertEqual(person[k], want[k])
+
+    def test_execute_queries(self):
+        project_id = bq_utils.app_identity.get_application_id()
+        dataset_id = bq_utils.get_combined_dataset_id()
+        self.assertIsNotNone(project_id)
+        self.assertIsNotNone(dataset_id)
+        test_util.delete_all_tables(dataset_id)
+
+        create_tables = ['person', 'observation']
+        for tbl in ['concept']:
+            if not bq_utils.table_exists(tbl, dataset_id=dataset_id):
+                create_tables.append(tbl)
+        for tbl in create_tables:
+            bq_utils.create_standard_table(tbl,
+                                           tbl,
+                                           dataset_id=dataset_id,
+                                           force_all_nullable=True)
+
+        gender_nonbinary_concept_id = 1585841
+        gender_nonbinary_source_concept_id = 123
+        sex_female_concept_id = 1585847
+        sex_female_source_concept_id = 45878463
+        for tmpl in INSERT_FAKE_PARTICIPANTS_TMPLS:
+            query = tmpl.render(
+                project_id=project_id,
+                dataset_id=dataset_id,
+                gender_concept_id=repopulate_person_post_deid.GENDER_CONCEPT_ID,
+                gender_nonbinary_concept_id=gender_nonbinary_concept_id,
+                gender_nonbinary_source_concept_id=
+                gender_nonbinary_source_concept_id,
+                sex_at_birth_concept_id=repopulate_person_post_deid.
+                SEX_AT_BIRTH_CONCEPT_ID,
+                sex_female_concept_id=sex_female_concept_id,
+                sex_female_source_concept_id=sex_female_source_concept_id)
+            try:
+                resp = bq_utils.query(query)
+            except HttpError as e:
+                self.fail("failed to execute query '{}': {}".format(
+                    query, e.content))
```

Comment:
Do you think it would be better to add these persons and codes in a setUp() and remove them/delete the tables in a tearDown()?

Reply:
It feels more likely to me that if I wrote a second test case, I'd probably want to use different data rather than the same data. Therefore while I moved table creation to setUp, I prefer to leave the row insertion here.